### PR TITLE
took whats in the 2nd attempt and copied them over to this repo

### PR DESCRIPTION
**Pull request by suchanlee**

### CODE REVIEW

**Comment by suchanlee** [comment]
you shouldn't create a new project to make changes or try something new. just create a git branch in the same repository, and create a pull request to it, like this.

File: To-List-Project-HTML2.html [hunk 1]
```diff
@@ -4,229 +4,30 @@
...
-		color: #FF8000;
-		font-size: 0.5em;
-	}
-	.accessories {
-		text-align: center;
-	}
-	
-	</style>
-	
+	<link href="todo.css" rel="stylesheet">
```
Comment (suchanlee) [inline To-List-Project-HTML2.html:7]: you're using 8 spaces for indenting, sometimes 4. Use either 2 or 4 spaces and be consistent with it.

Comment (suchanlee) [reply To-List-Project-HTML2.html:7]: Whitespace is very important for readability and it's important that you are consistent.

Comment (suchanlee) [reply To-List-Project-HTML2.html:7]: Also, always remove trailing spaces. https://www.google.com/search?q=trailing+spaces&oq=trailing+spaces&aqs=chrome..69i57j0l5.1791j0j1&sourceid=chrome&ie=UTF-8

A good text editor (e.g. sublime text, vscode) should have extensions allowing you to remove trailing whitespaces easily.

File: To-List-Project-HTML2.html [hunk 2]
```diff
@@ -4,229 +4,30 @@
...
+
 </head>
 
 
-<body onload = "load()">
+<body>
 	<div class= topBar>
```
Comment (suchanlee) [inline To-List-Project-HTML2.html:13]: Change `class= topBar` to `class="top-bar"`.
use quotes around the class name and use hyphens to separate words

Comment (suchanlee) [reply To-List-Project-HTML2.html:13]: Same for all your other class names

File: To-List-Project-HTML2.html [hunk 3]
```diff
@@ -4,229 +4,30 @@
...
-	
-</body>
-
 
+	<br>
 
+	<div class= dialog></div>
 
+	<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+  <script src="todo.js" type="text/javascript"></script>
```
Comment (suchanlee) [inline To-List-Project-HTML2.html:31]: this line and the above is what i mean by having inconsistent indenting. Use either 2 or 4 and be consistent

Comment (suchanlee) [reply To-List-Project-HTML2.html:31]: good job taking javascript code out into its own file. as well as css.

File: todo.css [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+.topBar {
+  font-family: sans-serif;
+  font-size: 1em;
+  text-align: center;
+  color: white;
+  position: fixed;
+  top:-3%;
+  left:0%;
+  z-parentListIndex: 100;
```
Comment (suchanlee) [inline todo.css:13]: never seen this before. do you mean `z-index`?

File: todo.css [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  background-color: #FAECA9
+}
+
+.topBar {
+  font-family: sans-serif;
+  font-size: 1em;
+  text-align: center;
+  color: white;
+  position: fixed;
+  top:-3%;
```
Comment (suchanlee) [inline todo.css:11]: this shouldn't be needed. you should put `padding: 0; margin: 0` on `body` instead.

File: todo.css [hunk 6]
```diff
@@ -0,0 +1,36 @@
+body {
+  background-color: #FAECA9
+}
+
+.topBar {
+  font-family: sans-serif;
```
Comment (suchanlee) [inline todo.css:6]: put this line on the `body` so that it gets propagated down to all the nodes, so that you don't have to put it for `.list` as well.

File: todo.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
+
+//DEFINITIONS
+var bullet = "- ";
+
+var button = $("<button>").attr("type", "submit").text("submit");
+var input = $("<input>").attr("type", "string");
+var label = $("<label>").append("Add an item <br>").append(this.input);
+
+var icon = function(src) {
```
Comment (suchanlee) [inline todo.js:9]: this should be `Icon` instead of `icon`

Comment (suchanlee) [reply todo.js:9]: Looks like the `src` parameter always has the same value. why should it be pass in instead of having it in the constructor?

File: todo.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+} else {
+  var items = savedData;
+  for (var i = 0; i < savedData.length; i++) {
+      var $p = $("<p>")
+      var deleteIcon = new icon("https://cdn3.iconfinder.com/data/icons/google-material-design-icons/48/ic_delete_48px-128.png");
+      var $deleteIcon = $("<img>")
+        .attr("src", deleteIcon.src)
+        .attr("alt", deleteIcon.alt)
+        .attr("width", deleteIcon.width)
+        .attr("style", String("opacity: " + deleteIcon.initialOpacity))
```
Comment (suchanlee) [inline todo.js:30]: dont need to wrap in `String()`

File: todo.js (hunk 7 above)
Comment (suchanlee) [reply todo.js:9]: actually you can just remove `src` and `alt` from `Icon` and just do `delete item`

File: todo.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  var items = [];
+} else {
+  var items = savedData;
+  for (var i = 0; i < savedData.length; i++) {
+      var $p = $("<p>")
+      var deleteIcon = new icon("https://cdn3.iconfinder.com/data/icons/google-material-design-icons/48/ic_delete_48px-128.png");
+      var $deleteIcon = $("<img>")
+        .attr("src", deleteIcon.src)
+        .attr("alt", deleteIcon.alt)
+        .attr("width", deleteIcon.width)
```
Comment (suchanlee) [inline todo.js:29]: you don't need style/width here, you can just put a class in the icon img element and style it in css right?

Comment (suchanlee) [reply todo.js:29]: In general, you should put styles in css, not in js.

File: todo.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+        .attr("style", String("opacity: " + deleteIcon.initialOpacity))
+        .on("mouseenter", function(event) {
```
Comment (suchanlee) [inline todo.js:31]: you dont need to do this for styles. instead if the img element had a class like `` and in css you had 
```
.delete-icon {
   opacity: 0.1;
}

.delete-icon:hover {
   opacity: 1;
}
```
you get the same thing and it takes styles outside of js

**Comment by suchanlee** [comment]
In general, I don't think too much has changed since the last version -- it's a bit cleaner but there's still lots of manual DOM manipulation that could get out of hand with larger apps. But for something simple like a todo list, it's okay.

I left comments regarding taking styles out of JS and having consistent whitespaces

**Review by seannam1218 — APPROVED** [review]
I have to write a review to approve the pull request?

File: To-List-Project-HTML2.html (hunk 1 above)
Comment (seannam1218) [reply To-List-Project-HTML2.html:7]: Noted!

File: To-List-Project-HTML2.html (hunk 2 above)
Comment (seannam1218) [reply To-List-Project-HTML2.html:13]: got it. Is this to differentiate between class names and variable names?

File: todo.css (hunk 6 above)
Comment (seannam1218) [reply todo.css:6]: got it

File: todo.css (hunk 5 above)
Comment (seannam1218) [reply todo.css:11]: you mean lines 10 and 11? So you mean change it so that the top bar doesn't stay on the screen when scrolling? Any particular reason for this?

File: To-List-Project-HTML2.html (hunk 3 above)
Comment (seannam1218) [reply To-List-Project-HTML2.html:31]: On my project in my computer, these have the same spacing. Maybe it's related to trailing white spaces? I'll keep an eye out for this issue

File: todo.css (hunk 4 above)
Comment (seannam1218) [reply todo.css:13]: my bad, this should be z-index

File: To-List-Project-HTML2.html (hunk 3 above)
Comment (suchanlee) [reply To-List-Project-HTML2.html:31]: it could be because you're using tabs instead of spaces. your editor should have the ability to make sure that your indents are using spaces instead of tabs

**Comment by suchanlee** [comment]
hmm i dont think so, you should branch off of this and create a pull request with the changes that I've mentioned, so I can review that

File: todo.js (hunk 7 above)
Comment (seannam1218) [reply todo.js:9]: When I made the Icon object constructor (? not sure about terminology), I was considering features like "edit item" button that may be implemented in the future, hence the src parameter.

File: todo.js (hunk 9 above)
Comment (seannam1218) [reply todo.js:29]: ahaaa got it.

File: todo.js (hunk 10 above)
Comment (seannam1218) [reply todo.js:31]: ohhhh ok. So a lot of animations are better to be coded in the css? It's a lot shorter for one

File: To-List-Project-HTML2.html (hunk 3 above)
Comment (seannam1218) [reply To-List-Project-HTML2.html:31]: omg r u serious?? this is just like the episode on Silicon Valley loll so there IS a difference between tabs and spaces...

Comment (seannam1218) [reply To-List-Project-HTML2.html:31]: so spaces are better than tabs? or vice versa?

File: To-List-Project-HTML2.html (hunk 2 above)
Comment (suchanlee) [reply To-List-Project-HTML2.html:13]: the quotes is I believe required, but some browsers will put them in for you. But things might not work for browsers that do not have quotes around class names.

For using hyphens to delimit words, I think it just is the industry standard for css class names

File: To-List-Project-HTML2.html (hunk 3 above)
Comment (suchanlee) [reply To-List-Project-HTML2.html:31]: yep like the episode on Silicon Valley...

Some people prefer tabs but a lot of people including me prefer spaces because it renders the same on every computer whereas tabs in some computers will be 2, in others 4, and in some others 8

File: todo.css (hunk 5 above)
Comment (suchanlee) [reply todo.css:11]: nope I meant just line 11. you need line 10 to keep the header fixed at top, but i think you don't need top: -3%

File: todo.js (hunk 7 above)
Comment (suchanlee) [reply todo.js:9]: ahh i see that makes sense.

File: todo.js (hunk 10 above)
Comment (suchanlee) [reply todo.js:31]: some animations require js, but there are a lot of things (like hover states) that are easier in css
